### PR TITLE
glsl-discard

### PR DESCRIPTION
**Pull request by shrekshao**

### CODE REVIEW

**Comment by pjcozzi** [comment]
Wow, this looks really good!

I made some style tweaks in ccafeab96ecc3f415eaabcf04c1122878a31b72d.  I will commit on lines in the commit to provide some rationale.  Given that many developers will be copying and pasting from the samples (which is the whole point!), we need to make sure they are clean, consistent, and easy to follow.

File: samples/glsl_discard.html [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexPosBuffer);
+        gl.vertexAttribPointer(positionLocation, 2, gl.FLOAT, false, 0, 0);
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+        
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexTexBuffer);
+        gl.vertexAttribPointer(texcoordLocation, 2, gl.FLOAT, false, 0, 0);
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+        
+        var img = new Image();
+        img.src = '../assets/img/Di-3d.png';
```
Comment (pjcozzi) [inline samples/glsl_discard.html]: @shrekshao did you create this image?  I want to make sure that we have the rights to redistribute it.

**Comment by pjcozzi** [comment]
I also added a helper for loading images in fc2f18f754c965d45b861f7f4fe0e5bfb334ef6f, which is slightly cleaner than using `Image` directly.

**Comment by pjcozzi** [comment]
@shrekshao @trungtle have you guys read [JavaScript: The Good Parts](http://www.amazon.com/JavaScript-Good-Parts-Douglas-Crockford/dp/0596517742)?  If not, I will order you a copy.

**Comment by pjcozzi** [comment]
That's all my comments.  Can you please make similar changes in the other branches (except `dev`, which I already updated), and then comment in the respective pull request so I know they are ready for review?  If you are able to make the changes soon, I will have time later today to review.

**Comment by shrekshao** [comment]
Yeah sure, i will take care of them after i getnup😂. Valuable advice by
experienced programmers!

File: samples/glsl_discard.html (hunk 1 above)
Comment (shrekshao) [inline samples/glsl_discard.html]: Yes, it's my work from the 3d modeling class, it should be fine.

**Comment by shrekshao** [comment]
I think there's a free pdf version
http://bdcampbell.net/javascript/book/javascript_the_good_parts.pdf

Yet I haven't started reading .... -_-! I don't know Trung's progress...

Yeah it must be helpful reading this during casual time.

**Comment by shrekshao** [comment]
Replace getAttribLocation call with layout location, done, tested.

**Comment by shrekshao** [comment]
Ready for a second review

**Comment by pjcozzi** [comment]
Great progress!  I will review this and all the other PRs tomorrow morning.

**Comment by pjcozzi** [comment]
Looks good!  I made some consistency improvements in 9604577d39b7786dc922ace8c97963c6917326ac.

**Pull request merged**